### PR TITLE
manta: init 1.6.0

### PR DESCRIPTION
**Pull request by jbedo**
###### Motivation for this change

manta: init 1.6.0

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by aanderse** [comment]
@jbedo I'd love to merge this... but after reading the description and github page I still have no idea what this software does :man_shrugging: 

Maybe if we call in some `python` experts like @FRidh and @jonringer they can review the package, ignoring anything about the software they don't know.

**Comment by jonringer** [comment]
python2 is already EOL, I don't think we should be using it in any new expressions

**Comment by jonringer** [comment]
however, it looks like they only support python2:
https://github.com/Illumina/manta/blob/75b5c38d4fcd2f6961197b28a41eb61856f2d976/src/python/libexec/denovo_scoring.py#L1
https://github.com/Illumina/manta/issues/180

File: pkgs/applications/science/biology/manta/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    rm makeRunScript.pyc
+    echo "import makeRunScript" | python2
+    popd
+    pushd $out/lib/python/pyflow
+    sed -i 's|#!/usr/bin/env python|#!${python2}/bin/python|' pyflow.py
+    sed -i 's|/bin/bash|${stdenv.shell}|' pyflowTaskWrapper.py
+    rm pyflowTaskWrapper.pyc pyflow.pyc
+    echo "import pyflowTaskWrapper" | python2
+    echo "import pyflow" | python2
+    popd
```
Comment (jonringer) [inline pkgs/applications/science/biology/manta/default.nix]: I think what you were intending to do was:

```
  postPatch = ''
    sed -i 's|/bin/bash|${stdenv.shell}|' lib/python/pyflow/pyflowTaskWrapper.py
  '';
  installCheck = true;
  installCheckPhase = ''
    rm $out/lib/python/**/*.pyc
    PYTHONPATH=$out/lib/python:$PYTHONPATH python -c 'import makeRunScript'
    PYTHONPATH=$out/lib/python/pyflow:$PYTHONPATH python -c 'import pyflowTaskWrapper; import pyflow'
  '';
```

**Comment by jonringer** [comment]
I'm fine with this usage of python2, it's just for some python integration, and upstream is already aware that it's an issue

File: pkgs/applications/science/biology/manta/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    repo = "manta";
+    rev = "v${version}";
+    sha256 = "1711xkcw8rpw9xv3bbm7v1aryjz4r341rkq5255192dg38sgq7w2";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ zlib python2 ];
+  postFixup = ''
+    pushd $out/lib/python
+    sed -i 's|/usr/bin/env python2|${python2}/bin/python|' makeRunScript.py
```
[the remark this answers is not in the document]
Comment (jbedo) [reply pkgs/applications/science/biology/manta/default.nix]: The way the software works is it generates a python script to manage the workflow which the user then executes. This patch is to correct the generated script.

File: pkgs/applications/science/biology/manta/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    rm makeRunScript.pyc
+    echo "import makeRunScript" | python2
```
[the remark this answers is not in the document]
Comment (jbedo) [reply pkgs/applications/science/biology/manta/default.nix]: Thanks!

File: pkgs/applications/science/biology/manta/default.nix (hunk 1 above)
Comment (jbedo) [reply pkgs/applications/science/biology/manta/default.nix]: That's much cleaner, thanks! I couldn't get installCheckPhase to execute though even with installCheck set to true, so i left the patching in postFixup. Is this ok?

Comment (jonringer) [reply pkgs/applications/science/biology/manta/default.nix]: `doInstallCheck = true;` is the correct attribute. I was tired :)

Comment (jbedo) [reply pkgs/applications/science/biology/manta/default.nix]: Thanks :)

File: pkgs/applications/science/biology/manta/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    owner = "Illumina";
+    repo = "manta";
+    rev = "v${version}";
+    sha256 = "1711xkcw8rpw9xv3bbm7v1aryjz4r341rkq5255192dg38sgq7w2";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ zlib python2 ];
+  postFixup = ''
+    sed -i 's|/usr/bin/env python2|${python2}/bin/python|' $out/lib/python/makeRunScript.py
```
Comment (FRidh) [inline pkgs/applications/science/biology/manta/default.nix]: ${python2.interpreter}

File: pkgs/applications/science/biology/manta/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    sed -i 's|/usr/bin/env python|${python2}/bin/python|' $out/lib/python/pyflow/pyflow.py
+    sed -i 's|/bin/bash|${stdenv.shell}|' $out/lib/python/pyflow/pyflowTaskWrapper.py
+  '';
+  doInstallCheck = true;
+  installCheckPhase = ''
+    rm $out/lib/python/**/*.pyc
+    PYTHONPATH=$out/lib/python:$PYTHONPATH python -c 'import makeRunScript'
```
Comment (FRidh) [inline pkgs/applications/science/biology/manta/default.nix]: `$out/lib/python` is not a typical site-packages folder. Why does it not install into `python2.sitePackages`?

**Review by FRidh — DISMISSED** [review]

File: pkgs/applications/science/biology/manta/default.nix (hunk 5 above)
Comment (jbedo) [reply pkgs/applications/science/biology/manta/default.nix]: Manta is not a python package, it is a c++ software with some supporting python code. They use python to manage the execution of the c++ core, i.e., the workflow, to allow computation on various types of computational resources (slurm/pbs clusters, cloud, direct, etc). The process is complicated and documented [here](https://github.com/Illumina/manta/blob/master/docs/userGuide/README.md#run-configuration-and-execution). I've verified complete execution on a non-trivial example.

File: pkgs/applications/science/biology/manta/default.nix (hunk 4 above)
Comment (jbedo) [reply pkgs/applications/science/biology/manta/default.nix]: Done.

File: pkgs/applications/science/biology/manta/default.nix (hunk 5 above)
Comment (jonringer) [reply pkgs/applications/science/biology/manta/default.nix]: the python scripts are more for utility, than being consumed by other python packages

Comment (jbedo) [reply pkgs/applications/science/biology/manta/default.nix]: Is this resolved?

Comment (jonringer) [reply pkgs/applications/science/biology/manta/default.nix]: Personally, I would say so

**Comment by jonringer** [comment]
would you need python at runtime? if so, we need to wrap the executable

```
        wrapProgram $i \
        --set JAVA_HOME "${jre}" \
        --prefix PATH : "${bash}/bin"
```
the package needed is `nativeBuildInputs = [ makeWrapper ]; `

**Comment by jbedo** [comment]
All references to python end up as absolute paths after the fixup phase, so a wrapper ends up unnecessary.

**Comment by jonringer** [comment]
@GrahamcOfBorg build manta

**Pull request merged**